From: Carlos Garnacho <carlosg@gnome.org>
Date: Sat, 23 Apr 2022 19:51:11 +0200
Subject: padOsd: Set menu manager on actions popup menu

Menus do need a menu manager to handle grabs for them, this
is more necessary now that the padOsd do also push a
Clutter.Grab.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2279>
Origin: upstream, 42.1, commit:1c9528391338d2f8b60fe790eb78e999c4950ad5
---
 js/ui/padOsd.js | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/js/ui/padOsd.js b/js/ui/padOsd.js
index 065df47..e23213c 100644
--- a/js/ui/padOsd.js
+++ b/js/ui/padOsd.js
@@ -144,6 +144,9 @@ var ActionComboBox = GObject.registerClass({
         this._editMenu.actor.hide();
         Main.uiGroup.add_actor(this._editMenu.actor);
 
+        this._editMenuManager = new PopupMenu.PopupMenuManager(this);
+        this._editMenuManager.addMenu(this._editMenu);
+
         this._actionLabels = new Map();
         this._actionLabels.set(GDesktopEnums.PadButtonAction.NONE, _("Application defined"));
         this._actionLabels.set(GDesktopEnums.PadButtonAction.HELP, _("Show on-screen help"));
